model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
+
+    def load(self, db, setting_name):
+        """Loads and returns the library's configuration setting
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :return: Any
+        """
+        integration = self._integration_owner.external_integration(db)
+        value = ConfigurationSetting.for_externalintegration(
+            setting_name,
+            integration).value
+
+        return value
+
+
+class CollectionConfigurationStorage(BaseConfigurationStorage):
+    """Serializes and deserializes values as library's configuration settings"""
+
+    def __init__(self, integration_owner, collection_id):
+        """Initializes a new instance of ConfigurationStorage class
+
+        :param integration_owner: External integration owner
+        :type integration_owner: CollectionExternalIntegrationOwner
+
+        :param collection_id: Collection's ID
+        :type collection_id: int
+        """
+        self._integration_owner = integration_owner
+        self._collection_id = collection_id
+
+    def save(self, db, setting_name, value):
+        """Save the value as as a new configuration setting
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :param value: Value to be saved
+        :type value: Any
+        """
+        integration = self._integration_owner.collection_external_integration(db, self._collection_id)
+        ConfigurationSetting.for_externalintegration(
+            setting_name,
+            integration).value = value
+
+    def load(self, db, setting_name):
+        """Loads and returns the library's configuration setting
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :return: Any
+        """
+        integration = self._integration_owner.collection_external_integration(db, self._collection_id)
+        value = ConfigurationSetting.for_externalintegration(
+            setting_name,
+            integration).value
+
+        return value
+
+
+class ConfigurationAttributeType(Enum):
+    """Enumeration of configuration setting types"""
+
+    TEXT = 'text'
+    TEXTAREA = 'textarea'
+    SELECT = 'select'
+    NUMBER = 'number'
+
+    def to_control_type(self):
+        """Converts the value to a attribute type understandable by circulation-web
+
+        :return: String representation of attribute's type
+        :rtype: string
+        """
+        # NOTE: For some reason, circulation-web converts "text" into <text> so we have to turn it into None
+        # In this case circulation-web will use <input>
```

Comment:
I looked into this but couldn't find the cause. @adriana-alter can you look into this? Is circulation-web creating a `<text>` tag (or maybe `<textarea>`?) when it hears about a configuration setting of type "text"?

Reply:
I suspect [this line](https://github.com/NYPL-Simplified/circulation-web/blob/c1eaead8a53dab419d760ee9af6f9eacdd6932ef/src/components/ProtocolFormField.tsx#L64)

Reply:
I believe so.  Should be no problem to fix, but, just to be totally clear: what do we want it to do instead?

Reply:
I think it should be `<input type="text">`

Reply:
Will do!  Can it go into the next sprint, or is it time-sensitive?

Reply:
I think next sprint should be fine. @vbessonov add a TODO to change this back after Adriana makes the change in circulation-web, file a ticket in JIRA and assign it to her.

Reply:
Sure, created [SIMPLY-3008](https://jira.nypl.org/browse/SIMPLY-3008) assigned it to Adriana and mentioned it in a TODO in the code.